barman/clients/cloud_backup.py

```diff
@@ -434,7 +454,14 @@ def parse_arguments(args=None):
         help="The name of the Azure resource group to which the compute instance and "
         "disks defined by the --snapshot-instance and --snapshot-disk arguments belong.",
     )
-    return parser.parse_args(args=args)
+
+    parsed_args = parser.parse_args(args=args)
+
+    # Perform mutual exclusivity check
+    if parsed_args.aws_snapshot_lock_duration is not None and parsed_args.aws_snapshot_lock_expiration_date is not None:
```

Comment:
Move and refactor this in the `_validate_config` function in this module.

Reply:
As this is only about mutually exclusiveness, we can also leverage of `argparse` to handle that for us. Something like this:

```python
    s3_lock_target_group = s3_arguments.add_mutually_exclusive_group()
    s3_lock_target_group.add_argument(
        "--aws-snapshot-lock-duration",
        help="The duration (in days) for which the snapshot should be locked. Must be between 1 and 36500. To lock a snapshopt, you must specify either this argument or --aws-snapshot-lock-expiration-date, but not both.",
        type=check_aws_snapshot_lock_duration_range,
    )
    s3_lock_target_group.add_argument(
        "--aws-snapshot-lock-expiration-date",
        help="The expiration date for a locked snapshot in the format YYYY-MM-DDThh:mm:ss.sssZ. To lock a snapshot, you must specify either this argument or --aws-snapshot-lock-duration, but not both.",
        type=check_timestamp
    )
```

If you attempt to specify both, `argparse` would throw an error:

```console
barman-cloud-backup: error: argument --aws-snapshot-lock-expiration-date: not allowed with argument --aws-snapshot-lock-duration
```